module/Makefile.in

```diff
@@ -90,7 +90,15 @@ modules_install-Linux:
 	if [ -n "$(DESTDIR)" ]; then \
 		find $$kmoddir -name 'modules.*' | xargs $(RM); \
 	fi
+	@# Debian ships tiny fake System.map files that are
+	@# syntactically valid but just say
+	@# "if you want system.map go install this package"
+	@# Naturally, depmod is less than amused by this.
+	@# So if we find it missing or with one of these present,
+	@# we check for the alternate path for the System.map
 	sysmap=$(INSTALL_MOD_PATH)/boot/System.map-@LINUX_VERSION@; \
+	([ -f "$$sysmap" ] && [$$(wc -l < "$$sysmap") -ge 100 ]) || \
```

Comment:
if `[$$(wc` works, it's by accident, and `{ [ -f "$$sysmap" ] && [ $$(wc -l < "$$sysmap") -ge 100 ]; } || \` avoids the needless subshell

Reply:
Always my favorite failure mode. "If that works, it shouldn't, and it should feel bad."

(Pushed, thanks.)

Reply:
I mean, it works if `wc -l` starts with spaces, which is the case on NetBSD, but not on coreutils or busybox so I don't think it ever works under Linux

Reply:
Curious.  I would have expected my testbed to have reached the second half of that statement and errored appropriately, as it is indeed Linux, and the map file does exist, so ?

Regardless, that should be good now, I believe?

Reply:
Yeah, it errors in that the `[1 -ge 100 ]` invocation fails as 127 => the subshell fails with 127 => it falls down to /u/l/d/b/S.m.

This is best illustrated by having permanent POSIX-induced brain damage and this being the first thing that pops into your head when reading this; failing that, however, an example:
```
$ sysmap=/boot/System.map-5.10.0-9-amd64 sh -c '([ -f "$sysmap" ] && [$(wc -l < "$sysmap") -ge 100 ]) || echo changing $?'
sh: 1: [1: not found
changing 127
```
For completeness: a comparison with NetBSD (well, anything since V5 AT&T UNIX) wc behaviour:
```
$ sysmap=/boot/System.map-5.10.0-9-amd64 sh -c '([ -f "$sysmap" ] && [$(echo " 1") -ge 100 ]) || echo changing $?'
changing 1
```

Yeah, the updated version is correct.

Reply:
Great.  Thanks for sorting it out, I'll get it merged.